File: codes/statisticalAnalysisLearning/statisticalLearning/FirstCourseMachineLearningCode/R/chapter8/gausssamp.R
gausssamp <- function(mu,sigma,N,sigmachol=FALSE){
  if(is.logical(sigmachol)){
    sigmachol = chol(sigma)
    sigmachol=t(sigmachol)
  }
  q = matrix(rnorm(nrow(mu)*N),nrow=nrow(mu),ncol=N)
  g = matrix(rep(mu,N),ncol=N)+(sigmachol%*%q)
  g = t(g)
  return(g)
}

File: codes/statisticalAnalysisLearning/statisticalLearning/FirstCourseMachineLearningCode/matlab/chapter8/gausssamp.m
function g = gausssamp(mu,sigma,N,sigmachol)

if ~exist('sigmachol')
    [sigmachol,p] = chol(sigma);
    sigmachol = sigmachol';
end
q = randn(length(mu),N);
g = repmat(mu,1,N)+ sigmachol*q;
g = g';